xikolo-common/ViewHelpers/LocalizedDateFormatter.swift

```diff
@@ -0,0 +1,20 @@
+//
+//  LocalizedDateFormatter.swift
+//  xikolo-ios
+//
+//  Created by Bjarne Sievers on 19.09.17.
+//  Copyright © 2017 HPI. All rights reserved.
+//
+
+import Foundation
+
+class LocalizedDateFormatterFactory {
+    class func newDateFormatter(with locale: Locale = Locale.current, and calendar: Calendar = Calendar.current, and timeZone: TimeZone = TimeZone.current) -> DateFormatter {
```

Comment:
That's looking more like Java-style programming.
Please change this to an extension of `DateFormatter` and move this file to the categories directory.
```swift
extension DateFormatter {
    static localizedFormatter( ...
}
```
For this function, I do not really like the usage of "with" and "and". I think this makes it harder to understand the purpose of all three parameters. Just use "locale", "calender" and "timeZone" and no _short label_.

Additonally, please use `autoupdatingCurrent ` instead of `current`.

Reply:
Yeah, I don't really like them either. 

About the `autoupdatingCurrentLocale`:
>**Note:** In theory you could use an auto-updating locale (autoupdatingCurrentLocale) to create a locale that automatically accounts for changes in the user’s locale settings. In practice this currently does not work with date formatters."

Found [here](https://developer.apple.com/library/content/documentation/Cocoa/Conceptual/DataFormatting/Articles/dfDateFormatting10_4.html) from apple

Reply:
Oh snap. Thanks Apple.
This currently does not work with date formatters, but it might work in futures releases. So I would use the auto-updating version.

Reply:
Okay I agree that the localized formatter should be an extension, but preconfigured ones for the table view cells with date styles set? I'm not so sure about that. I thought about extensions as the possibility to add common functionality back into the class. But with the cell formatters it would be quite a custom configuration. What do you think?

Reply:
I would not try to save a few lines of code by providing common date formatters via an extension. Just return a localized formatter via an extension. For the cell configuration, store the formatter as a property of the view controller.